user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleBatchInformationSpreadsheet.java

```diff
@@ -299,6 +301,82 @@ public String toString() {
           .add("customerComment='" + customerComment + "'")
           .toString();
     }
+
+    @Override
+    public SampleInfo clone() {
+      try {
+        SampleInfo clone = (SampleInfo) super.clone();
+        clone.analysisToBePerformed = analysisToBePerformed;
+        clone.analyte = analyte;
+        clone.biologicalReplicate = biologicalReplicate;
+        clone.customerComment = customerComment;
+        clone.experimentalGroup = experimentalGroup;
+        clone.sampleCode = sampleCode;
+        clone.sampleId = sampleId;
+        clone.sampleLabel = sampleLabel;
+        clone.species = species;
+        clone.specimen = specimen;
+        return clone;
+      } catch (CloneNotSupportedException e) {
+        throw new AssertionError();
+      }
+    }
```

Comment:
[Joshua Bloch](https://www.artima.com/articles/josh-bloch-on-design#part13) advises against implementing the clone interface and defers to rather implementing a copy constructor.

Reply:
Yes, if you have a different possibility let me know :D

Reply:
How about the one suggested [here](https://www.baeldung.com/java-copy-constructor):